More psi4 fixes

alesgenova opened the pull request.

berquist commented on cclib/parser/psi4parser.py:
```diff
@@ -115,7 +122,7 @@ def extract(self, inputfile, line):
         # ...
         #
-        if (self.section == "Geometry") and ("Geometry (in Angstrom), charge" in line):
+        if (self.section == "Geometry") and ("Geometry (in Angstrom), charge" in line) and not hasattr(self, 'finite_difference'):
```
Can you put the `and not ...` on a new line?

berquist commented on cclib/parser/psi4parser.py:
```diff
@@ -174,7 +181,7 @@ def extract(self, inputfile, line):
 
         # The printout for Psi4 has a more obvious trigger for the SCF parameter printout.
-        if (self.section == "Algorithm") and (line.strip() == "==> Algorithm <=="):
+        if (self.section == "Algorithm") and (line.strip() == "==> Algorithm <==") and not hasattr(self, 'finite_difference'):
```
Same as above.

berquist commented on cclib/parser/psi4parser.py:
```diff
@@ -371,7 +378,7 @@ def get_normalization_factor(exp, lx, ly, lz):
         # one for the density-fitted algorithm and one for the exact one, and the iterations are
         # printed in two blocks separated by some set-up information.
-        if (self.section == "Iterations") and (line.strip() == "==> Iterations <=="):
+        if (self.section == "Iterations") and (line.strip() == "==> Iterations <==")  and not hasattr(self, 'finite_difference'):
```
Same as above.

berquist commented on cclib/parser/psi4parser.py:
```diff
@@ -654,7 +662,7 @@ def get_normalization_factor(exp, lx, ly, lz):
         #  ---------------------------------------------------------------------------------------------
         #
-        if (self.section == "Convergence Check") and line.strip() == "==> Convergence Check <==":
+        if (self.section == "Convergence Check") and line.strip() == "==> Convergence Check <=="  and not hasattr(self, 'finite_difference'):
```
Same as above.

berquist commented on cclib/parser/psi4parser.py:
```diff
@@ -102,6 +102,13 @@ def extract(self, inputfile, line):
             if self.reference[0] == 'C':
                 self.reference = self.reference[1:]
 
+        # Parse the XC density functional
```
Are you using `metadata` for anything? If so, would be willing to offer us input on what you'd like it to contain? We have multiple running issues: https://github.com/cclib/cclib/issues/451, https://github.com/cclib/cclib/issues/523, https://github.com/cclib/cclib/pull/599.

langner commented:
Are there any additional logfiles you could contribute to test this code in regressions?

alesgenova commented:
@berquist 
I addressed your comments, it looks like travis is passing again.

@langner 
I am using some fairly basic input/output at the moment, but I will contribute some additional outputs in the future (together with some proper checks on the values of parsed output)

langner commented:
> @berquist
> I addressed your comments, it looks like travis is passing again.

Looks good. If you're up for it, please rebase that commit with a more useful message.

> @langner
> I am using some fairly basic input/output at the moment, but I will contribute some additional outputs in the future (together with some proper checks on the values of parsed output)

I understand. If that basic output is not a unit test logfile from our repo, it would be awesome if you could donate it to our regression suite. Even simple example outputs are great for making sure bugs don't come back.

alesgenova commented:
@langner 
Reworded the commit message!
